lib/CPathfinder.cpp

```diff
@@ -451,53 +645,40 @@ void CGPath::convert( ui8 mode )
 	}
 }
 
-CPathsInfo::CPathsInfo( const int3 &Sizes )
-:sizes(Sizes)
+CPathsInfo::CPathsInfo(const int3 &Sizes)
+	: sizes(Sizes)
 {
 	hero = nullptr;
-	nodes = new CGPathNode**[sizes.x];
+	nodes.resize(boost::extents[sizes.x][sizes.y][sizes.z][ELayer::NUM_LAYERS]);
```

Comment:
Thanks for explanation!

I'm not yet worked on any optimizations until all rules / limitations are done and working, but I almost completed with them so I'll jump on it soon.